samples/model-builder/deploy_model_sample.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from google.cloud import aiplatform
+from typing import Optional, Sequence, Tuple, Dict
+from google.cloud.aiplatform import explain
+
+#  [START aiplatform_sdk_deploy_model_sample]
+def deploy_model_sample(
```

Comment:
I'm assuming this is for models with automatic resources as machine type is not exposed. If that's the case, I think the following arguments can be removed but please confirm:

`accelerator_type`
`accelerator_count`
`explanation_metadata`
`explanation_parameters`

Reply:
Yes it's for automatic resources.

I checked the internal deploy function and it looks like `accelerator_type` and `accelerator_count` if `machine-type` is not provided. However, even if it intuitively makes sense, I would argue that the docstrings should make their co-dependence clear. 

I would also recommend a refactor to bundle `machine_type`, `accelerator_type` and `accelerator_count` together, since they can't be used independently. If you want to keep the client-facing API flat, that's fine but in private functions it will be cleaner bundled.

From looking at `_deploy_call`, I don't see `explanation_metadata` and `explanation_parameters` relying on `machine_type` being set. Is there some restriction from the service the requires `machine_type?

Reply:
Agree on the refactor and updating the docstring. Only custom models take `explanation_metadata` and `explanation_parameters` as AutoMLs support it natively (if supported) and machine_type is required from custom models.

Reply:
@vinnysenthil to confirm

Reply:
@sasha-gitg Logged a bug: https://b.corp.google.com/issues/186121905